src/k8s/pkg/k8sd/setup/k8s_apiserver_proxy_test.go

```diff
@@ -1,8 +1,91 @@
 package setup_test
 
-import "testing"
+import (
+	"encoding/json"
+	"os"
+	"path"
+	"testing"
 
-func TestK8sAPIServerProxy(t *testing.T) {
-	// TODO: setup a mock snap with paths, then verify setup.K8sAPIServerProxy().
-	// See TestContainerd() for inspiration.
+	"github.com/canonical/k8s/pkg/k8sd/setup"
+	"github.com/canonical/k8s/pkg/proxy"
+	"github.com/canonical/k8s/pkg/snap/mock"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/k8s/pkg/utils"
+	. "github.com/onsi/gomega"
+)
+
+func setK8sApiServerMock(s *mock.Snap, dir string) {
+	s.Mock = mock.Mock{
+		ServiceArgumentsDir:   path.Join(dir, "args"),
+		ServiceExtraConfigDir: path.Join(dir, "args/conf.d"),
+	}
+}
+
+func TestK8sApiServerProxy(t *testing.T) {
+	t.Run("Args", func(t *testing.T) {
+		g := NewWithT(t)
+
+		s := mustSetupSnapAndDirectories(t, setK8sApiServerMock)
+
+		g.Expect(setup.K8sAPIServerProxy(s, []string{}))
```

Comment:
This is not right. `g.Expect()` does not validate anything. You need to to add `.To(...)`